test/cri-containerd/pod_update_test.go

```diff
@@ -0,0 +1,234 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+func Test_Pod_UpdateResources_Memory(t *testing.T) {
+	type config struct {
+		name             string
+		requiredFeatures []string
+		runtimeHandler   string
+		sandboxImage     string
+	}
+	tests := []config{
+		{
+			name:             "WCOW_Hypervisor",
+			requiredFeatures: []string{featureWCOWHypervisor},
+			runtimeHandler:   wcowHypervisorRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+		},
+		{
+			name:             "LCOW",
+			requiredFeatures: []string{featureLCOW},
+			runtimeHandler:   lcowRuntimeHandler,
+			sandboxImage:     imageLcowK8sPause,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, []string{test.sandboxImage})
+			} else {
+				pullRequiredImages(t, []string{test.sandboxImage})
+			}
+			var startingMemorySize int64 = 768 * 1024 * 1024
+			podRequest := &runtime.RunPodSandboxRequest{
+				Config: &runtime.PodSandboxConfig{
+					Metadata: &runtime.PodSandboxMetadata{
+						Name:      t.Name(),
+						Namespace: testNamespace,
+					},
+					Annotations: map[string]string{
+						"io.microsoft.container.memory.sizeinmb": fmt.Sprintf("%d", startingMemorySize),
+					},
+				},
+				RuntimeHandler: test.runtimeHandler,
+			}
+
+			client := newTestRuntimeClient(t)
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			podID := runPodSandbox(t, client, ctx, podRequest)
+			defer removePodSandbox(t, client, ctx, podID)
+			defer stopPodSandbox(t, client, ctx, podID)
+
+			// make request for shrinking memory size
+			newMemorySize := startingMemorySize / 2
+			updateReq := &runtime.UpdateContainerResourcesRequest{
+				ContainerId: podID,
+			}
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				updateReq.Linux = &runtime.LinuxContainerResources{
+					MemoryLimitInBytes: newMemorySize,
+				}
+			} else {
+				updateReq.Windows = &runtime.WindowsContainerResources{
+					MemoryLimitInBytes: newMemorySize,
+				}
+			}
+
+			if _, err := client.UpdateContainerResources(ctx, updateReq); err != nil {
+				t.Fatalf("updating container resources for %s with %v", podID, err)
+			}
+		})
```

Comment:
Are we not doing any verification for pod updates?

Reply:
Unfortunately that's really difficult to do. Sorry, I meant to post in the PR about it. We really need to be able to query HCS for accurate information. There's a functional test for verifying memory updates and when those tests are runnable again, I'll add something for verifying cpu as well. So for now, the cri-containerd tests for pod updates will just verify nothing breaks :)

Reply:
Sounds good just wanted to check. thanks.